pin-project 0.3.3 introduces breaking changes about "missing extern crate core" compared to 0.3.2

### 0.3.2

```
$ cargo update -p pin-project --precise 0.3.2
    Updating crates.io index
    Updating pin-project v0.3.3 -> v0.3.2

$ cargo test
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running target/debug/deps/futures_compat-27fb39e639db3ee7

running 1 test
test implements_error ... ok
```

### 0.3.3

```
$ cargo update -p pin-project --precise 0.3.3
    Updating crates.io index
    Updating pin-project v0.3.2 -> v0.3.3

$ cargo test
   Compiling snafu v0.4.1 (/Users/shep/Projects/snafu)
error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_future.rs:127:18
    |
127 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_future.rs:166:18
    |
166 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_stream.rs:129:18
    |
129 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_stream.rs:171:18
    |
171 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0433`.
error: Could not compile `snafu`.
```

--- a/src/unsafe_project/mod.rs
+++ b/src/unsafe_project/mod.rs
@@ -1,6 +1,6 @@
-use proc_macro2::{Ident, Span, TokenStream};
-use quote::quote_spanned;
-use syn::{spanned::Spanned, Generics, Item, Result, Type};
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::{Generics, Item, Result, Type};
 
 mod enums;
 mod structs;
@@ -29,27 +29,20 @@ fn proj_generics(generics: &Generics) -> Generics {
 // conditional Unpin implementation
 
 #[derive(Default)]
-struct ImplUnpin(
-    Option<(
-        // generics
-        Generics,
-        // span
-        Span,
-    )>,
-);
+struct ImplUnpin(Option<Generics>);
 
 impl ImplUnpin {
     /// Parses attribute arguments.
     fn new(args: TokenStream, generics: &Generics) -> Result<Self> {
         match &*args.to_string() {
             "" => Ok(Self::default()),
-            "Unpin" => Ok(Self(Some((generics.clone(), args.span())))),
+            "Unpin" => Ok(Self(Some(generics.clone()))),
             _ => Err(error!(args, "an invalid argument was passed")),
         }
     }
 
     fn push(&mut self, ty: &Type) {
-        if let Some((generics, _)) = &mut self.0 {
+        if let Some(generics) = &mut self.0 {
             generics
                 .make_where_clause()
                 .predicates
@@ -60,9 +53,9 @@ impl ImplUnpin {
     /// Creates `Unpin` implementation.
     fn build(self, ident: &Ident) -> TokenStream {
         self.0
-            .map(|(generics, span)| {
+            .map(|generics| {
                 let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
-                quote_spanned! { span =>
+                quote! {
                     impl #impl_generics ::core::marker::Unpin for #ident #ty_generics #where_clause {}
                 }
             })

--- a/tests/ui/unsafe_project/unpin_conflict.stderr
+++ b/tests/ui/unsafe_project/unpin_conflict.stderr
@@ -1,26 +1,26 @@
 error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Foo<_, _>`:
-  --> $DIR/unpin_conflict.rs:10:18
+  --> $DIR/unpin_conflict.rs:10:1
    |
 10 | #[unsafe_project(Unpin)] //~ ERROR E0119
-   |                  ^^^^^ conflicting implementation for `Foo<_, _>`
+   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Foo<_, _>`
 ...
 26 | impl<T, U> Unpin for Foo<T, U> where T: Unpin {} // Conditional Unpin impl
    | --------------------------------------------- first implementation here
 
 error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Bar<_, _>`:
-  --> $DIR/unpin_conflict.rs:30:18
+  --> $DIR/unpin_conflict.rs:30:1
    |
 30 | #[unsafe_project(Unpin)] //~ ERROR E0119
-   |                  ^^^^^ conflicting implementation for `Bar<_, _>`
+   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Bar<_, _>`
 ...
 46 | impl<T, U> Unpin for Bar<T, U> {} // Non-conditional Unpin impl
    | ------------------------------ first implementation here
 
 error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `Baz<_, _>`:
-  --> $DIR/unpin_conflict.rs:48:18
+  --> $DIR/unpin_conflict.rs:48:1
    |
 48 | #[unsafe_project(Unpin)] //~ ERROR E0119
-   |                  ^^^^^ conflicting implementation for `Baz<_, _>`
+   | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Baz<_, _>`
 ...
 64 | impl<T: Unpin, U: Unpin> Unpin for Baz<T, U> {} // Conditional Unpin impl
    | -------------------------------------------- first implementation here

